### PR TITLE
A psychotic NPC - The Sheep Counter

### PR DESCRIPTION
**Pull request by Brytenwalda**
#### Summary
```SUMMARY:  Content "Adds a new psychotic NPC to this mad world."```

#### Purpose of change

This post-apocalyptic world needs more psychotic NPCs and this is the first one I'd like to add. A mad hunter who spends his days counting imaginary sheep, believing that doing so will save the world. Well-meaning and generally good. He's also a mission-giver.

#### Describe the solution

This character has several hand-written and unique dialogue options and is intended to be a humorous addition to this dark world. I hope it can add some colour and uniqueness to the game. 

#### Additional context

I'm only not sure how to actually make this character spawn. I think it would be a good idea to make him spawn in a wood cabin, as he's a hunter/fisherman.

### CODE REVIEW

File: data\json\npcs\PS_SHEEP [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    "dynamic_line": "The world ended because at some point we got comfortable enough to stop paying attention to the delicate 'sheep equilibrium'.",
+    "responses": [
+      { "text": "Sheep equilibrium?", "topic": "TALK_SHEEP_KNOW" },
+      { "text": "Okay, I'm done.", "topic": "TALK_DONE" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_SHEEP_KNOW",
+    "dynamic_line": "Yes. They hold the secret. The world rests on a very delicate equilibrium and the key is to count them, at all times. I keep a detailed log in my head... we may still be able to undo everything! When the time is right, I'll publish a paper, reveal the secret to the broader scientific community and we'll work together to stop the cataclysm.",
```
Comment (Qrox) [inline data\json\npcs\PS_SHEEP]:
```suggestion
    "dynamic_line": "Yes.  They hold the secret.  The world rests on a very delicate equilibrium and the key is to count them, at all times.  I keep a detailed log in my head… we may still be able to undo everything!  When the time is right, I'll publish a paper, reveal the secret to the broader scientific community and we'll work together to stop the cataclysm.",
```

File: data\json\npcs\PS_SHEEP [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+      { "text": "Oh, okay.", "topic": "TALK_SHEEP" },
+	  { "text": "You do realise that there is no scientific community to speak of anymore, right?", "topic": "TALK_SHEEP_SCIENCE" }
+      { "text": "Okay, I'm done. See ya.", "topic": "TALK_DONE" }
+    ]
+  },
+  
+     {
+    "type": "talk_topic",
+    "id": "TALK_SHEEP_SCIENCE",
+    "dynamic_line": "There is a Hub of world's greatest minds out there. They told me to keep counting for now, and so I shall. I do not intend to let them down, I must keep a very detailed log, in my head, so no one can take it from me.",
```
Comment (Qrox) [inline data\json\npcs\PS_SHEEP]:
```suggestion
    "dynamic_line": "There is a Hub of world's greatest minds out there.  They told me to keep counting for now, and so I shall.  I do not intend to let them down, I must keep a very detailed log, in my head, so no one can take it from me.",
```

File: data\json\npcs\PS_SHEEP [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+      { "text": "Okay, I'm done.", "topic": "TALK_DONE" }
+    ]
+  },
+  
+  {
+      "type": "talk_topic",
+    "id": "TALK_SHEEP_MANY",
+    "dynamic_line": "Sixty-eight white, eleven brown, eight black, three red, two pink and one blue. See? Very detailed.",
+    "responses": [
+      { "text": "That's... fascinating.", "topic": "TALK_SHEEP" },
```
Comment (Qrox) [inline data\json\npcs\PS_SHEEP]:
```suggestion
      { "text": "That's… fascinating.", "topic": "TALK_SHEEP" },
```

File: data\json\npcs\PS_SHEEP [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    "responses": [
+      { "text": "How many sheep have you counted so far?", "topic": "TALK_SHEEP_MANY" },
+      { "text": "Okay, I'm done.", "topic": "TALK_DONE" }
+    ]
+  },
+  
+  {
+      "type": "talk_topic",
+    "id": "TALK_SHEEP_MANY",
+    "dynamic_line": "Sixty-eight white, eleven brown, eight black, three red, two pink and one blue. See? Very detailed.",
```
Comment (Qrox) [inline data\json\npcs\PS_SHEEP]:
```suggestion
    "dynamic_line": "Sixty-eight white, eleven brown, eight black, three red, two pink and one blue.  See?  Very detailed.",
```

File: data\json\npcs\PS_SHEEP [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+    "responses": [
+      { "text": "That's... fascinating.", "topic": "TALK_SHEEP" },
+      { "text": "Okay, I'm done.", "topic": "TALK_DONE" }
+    ]
+  },
+  
+  {
+      "type": "talk_topic",
+    "id": "TALK_SHEEP_WHY",
+    "dynamic_line": "I'm counting sheep. I'm also hunting and fishing around these parts. But I never kill sheep. I just count them. Very important. You can kill, I can't, but I must count. Must. Always.",
```
Comment (Qrox) [inline data\json\npcs\PS_SHEEP]:
```suggestion
    "dynamic_line": "I'm counting sheep.  I'm also hunting and fishing around these parts.  But I never kill sheep.  I just count them.  Very important.  You can kill, I can't, but I must count.  Must.  Always.",
```

File: data\json\npcs\PS_SHEEP [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+      { "text": "Great.", "topic": "TALK_SHEEP" },
+      { "text": "Okay, I'm done.", "topic": "TALK_DONE" }
+    ]
+  },
+  
+  
+      {
+      "type": "talk_topic",
+    "id": "TALK_SHEEP_IDIOT",
+    "dynamic_line": "Stupid is as stupid does. I guess you just can't grasp the secret. I don't blame you, takes a titanic intellect to do so.",
```
Comment (Qrox) [inline data\json\npcs\PS_SHEEP]:
```suggestion
    "dynamic_line": "Stupid is as stupid does.  I guess you just can't grasp the secret.  I don't blame you, takes a titanic intellect to do so.",
```

File: data\json\npcs\PS_SHEEP [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    "responses": [
+      { "text": "Of course.", "topic": "TALK_SHEEP" },
+	  { "text": "There are... no words.", "topic": "TALK_DONE" },
```
Comment (Qrox) [inline data\json\npcs\PS_SHEEP]:
```suggestion
	  { "text": "There are… no words.", "topic": "TALK_DONE" },
```

File: data\json\npcs\PS_SHEEP [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+	  { "text": "Uh, huh.", "topic": "TALK_SHEEP" },
+      { "text": "Okay, I'm done.", "topic": "TALK_DONE" }
+    ]
+  },
+  
+    {
+    "type": "talk_topic",
+    "id": "TALK_SHEEP_BYE",
+    "dynamic_line": "I'll be counting.",
+    "responses": [ { "text": "...", "topic": "TALK_DONE" } ]
```
Comment (Qrox) [inline data\json\npcs\PS_SHEEP]:
```suggestion
    "responses": [ { "text": "…", "topic": "TALK_DONE" } ]
```

File: data\json\npcs\PS_SHEEP [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+    "goal": "MGOAL_FIND_ITEM",
+    "difficulty": 2,
+    "value": 800,
+    "item": "bolt_steel",
+    "count": 50,
+    "origins": [ "ORIGIN_SECONDARY" ],
+	"end": { "effect": [ { "u_buy_item": "RobofacCoin", "count": 4 } ] },
+    "dialogue": {
+      "describe": "I'm running low on bolts for my crossbow.",
+      "offer": "I've been hunting for a while now, and good quality steel bolts are really hard to come by these days. Could you bring me, let's say, 50 steel bolts? You'd be a real lifesaver. I have some Hub coins for your troubles.",
```
Comment (Qrox) [inline data\json\npcs\PS_SHEEP]:
```suggestion
      "offer": "I've been hunting for a while now, and good quality steel bolts are really hard to come by these days.  Could you bring me, let's say, 50 steel bolts?  You'd be a real lifesaver.  I have some Hub coins for your troubles.",
```

File: data\json\npcs\PS_SHEEP [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+      "accepted": "Thanks, it's good to know that kindness is not gone from this world.",
+      "rejected": "That's fine, you cannot expect a stranger to care.",
+      "advice": "There must be some hunting gear in survival shops, but towns can be very dangerous these days.",
+      "inquire": "Do you have the bolts?",
+      "success": "Thanks! You're absolutely amazing!",
+      "success_lie": "OK, then hand them over.",
+      "failure": "That's fine. I know it's not an easy task."
```
Comment (Qrox) [inline data\json\npcs\PS_SHEEP]:
```suggestion
      "failure": "That's fine.  I know it's not an easy task."
```

**Comment by Night-Pryanik** [comment]
Please [lint](http://dev.narc.ro/cataclysm/format.html) your changes.

File: data\json\npcs\PS_SHEEP [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+  
+  {
+    "id": "MISSION_SHEEP_GATHER",
+    "type": "mission_definition",
+    "name": "I need new bolts for my crossbow.",
+    "description": "Help the odd hunter by finding 50 steel bolts for his crossbow.",
+    "goal": "MGOAL_FIND_ITEM",
+    "difficulty": 2,
+    "value": 800,
+    "item": "bolt_steel",
```
Comment (anothersimulacrum) [inline data\json\npcs\PS_SHEEP]: These are the non-craftable aluminun broadhead bolts - they aren't steel bolts (despite the id).

**Comment by davidpwbrown** [comment]
as with #35121 please add a new faction or use no_faction faction

**Comment by I-am-Erk** [comment]
@Brytenwalda I'd love to merge this and your other changes, they just need a few fixes and some conflict resolution. Feel free to reopen these if you come back to fix up the last bits. I can talk to you on discord to help fix the last bits if you need help. In this case, it's a single line change and I'd fix it if maintainers were allowed to make edits.

**Pull request closed without merging**